### PR TITLE
Add more detail to dns/secure event docs; confirm it with test cases.

### PR DESCRIPTION
**Pull request by dave-r12**
I'm hopeful this makes the docs and behavior more clear. Let me know if it muddies the waters.

### CODE REVIEW

File: okhttp-tests/src/test/java/okhttp3/EventListenerTest.java [hunk 1]
```diff
@@ -95,17 +98,70 @@
...
+    assertFalse(recordedEvents.contains(DnsEnd.class));
+  }
+
+  @Test public void multipleDnsLookupsForSingleCall() throws IOException {
+    server.enqueue(new MockResponse()
+        .setResponseCode(301)
+        .setHeader("Location", "http://www.fakeurl:" + server.getPort()));
+    server.enqueue(new MockResponse());
+
+    FakeDns dns = new FakeDns();
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: Nice

File: okhttp-tests/src/test/java/okhttp3/EventListenerTest.java [hunk 2]
```diff
@@ -147,9 +203,9 @@
     } catch (IOException expected) {
     }
 
-    listener.findNextEvent(DnsStart.class);
+    assertNotNull(listener.removeUpToEvent(DnsStart.class));
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: should removeUpToEvent() throw if it isn’t found?

Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: ... at least in tests I assume we’ll always know whether an event should be present?

Comment (dave-r12) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: Ahh yeah, that's probably a safer approach.

**Pull request merged**